### DISCUSSION TITLE
Integration guide for Supabase

### DISCUSSION THREAD

**Discussion by GrimLothar** [Web3Auth/Web3Auth#582] [Help]
Hey! I noticed that web3auth works perfectly with Firebase, and saw the guide to make it work.
Was wondering if the same could be done with Supabase? 

https://supabase.com/

In case you don't know Supabase, it's an open source alternative to Firebase, which uses a relational Postgre DB instead of the document type approach that Firebase uses. 

Thank you!

**Comment by shahbaz17** [comment]
Hey @grimAgent 

Yes, it can be as far as it returns an idToken in JWT format. Have you given a try? Simply replace the Firebase part and get the idToken from Supabase and pass that to web3auth.

But, one thing to note would be; the JWKS endpoint. Does Supabase exposes its JWKS to an endpoint? This is very much needed to verify the idToken at our end.

**Reply by shahbaz17** [reply]
Also, looked into https://github.com/supabase/supabase/discussions/5911 and asymmetric keys are not supported atm by Supabase.

**Reply by GrimLothar** [reply]
Yes, that's what I saw as well, but I wasn't sure that's what it meant. I also opened a discussion on their side to see if it's possible:
https://github.com/supabase/supabase/discussions/8883
